galsim/correlatednoise.py

```diff
@@ -729,27 +751,20 @@ def _get_update_rootps(self, shape, wcs):
         # Query using the rfft2/irfft2 half-sized shape (shape[0], shape[1] // 2 + 1)
         half_shape = (shape[0], shape[1] // 2 + 1)
         for rootps_array, saved_wcs in self._rootps_store:
-            if rootps_array.shape == half_shape:
-                if ( (wcs is None and saved_wcs.isPixelScale() and saved_wcs.scale == 1.) or
-                     wcs == saved_wcs ):
-                    use_stored = True
-                    rootps = rootps_array
-                    break
+            if rootps_array.shape == half_shape and wcs == saved_wcs:
+                use_stored = True
+                rootps = rootps_array
+                break
 
         # If not, draw the correlation function to the desired size and resolution, then DFT to
         # generate the required array of the square root of the power spectrum
         if use_stored is False:
-            newcf = galsim.ImageD(shape[1], shape[0]) # set the corr func to be the correct size
-            # set the wcs...
-            if wcs is None:
-                newcf.scale = 1.
-            else:
-                newcf.wcs = wcs
-            # Then draw this correlation function into an array.  If this is not done at the same
-            # wcs as the original image from which the CF derives, even if the image is rotated,
-            # then this step requires interpolation and the newcf (used to generate the PS below) is
-            # thus approximate at some level
-            newcf = self.drawImage(newcf)
+            # Draw this correlation function into an array.  If this is not done at the same wcs as
+            # the original image from which the CF derives, even if the image is rotated, then this
+            # step requires interpolation and the newcf (used to generate the PS below) is thus
+            # approximate at some level
```

Comment:
I believe the way it is implemented, if you `addNoise` onto an image that has the same WCS as the one you used when you built the CF, then the effects of the WCS should cancel out.  I.e. it would be the same as if you set the WCS to be `PixelScale(1.)`.  

I _think_ my unit tests for uncorrelated noise at least verify that this is the case if the `CorrelatedNoise` object doesn't actually have any correlations.  I don't know quite how to structure a test to confirm that this is working in the more general case.  If you have ideas for a unit test, I'd be happy to add it.

In any case, the bit that I thought mattered more was the whitening option, where you don't typically have the same WCS in the input and output images.  Currently, we just use a pixel scale difference for RealGalaxy, but I could imagine caring about the slightly non-square pixels in the HST detector or the output image.  Again, if you have an idea for a unit test of this, please let me know.